crates/ra_syntax/src/parsing/lexer.rs

```diff
@@ -16,51 +15,63 @@ use crate::{
 pub struct Token {
     /// The kind of token.
     pub kind: SyntaxKind,
-    /// The length of the token.
-    pub len: TextSize,
+    /// The offset of the token.
+    pub offset: TextSize,
```

Comment:
This would prevent any future incremental tokenization, as a adding or removing a single character would require updating all tokens after it.

Reply:
These tokens are one-off we create them for a moment and then throw off like we never had them.
And I don't think we plan to store these tokens with their offsets in the future too

Reply:
@bjorn3 is correct, we deliberately avoid using offsets if we can help it, including this case.